hm_1/tasks/task_01_01.py

```diff
@@ -0,0 +1,2 @@
+def check_power_of_2(a: int) -> bool:
+    return bool((a & (a - 1)) == 0 and a != 0)
```

Comment:
Can you explain how this magic works?

Reply:
We use operator & for multiply numbers by bits, for example, we have 16, we know that it is a power of 2 and want to check this, in my opinion, the simplest way is to write the number in base =2 and number -1 in base 2 (in our example 10000-16, 1111-15, we multiply it and it gives as zero, for other numbers, which isn't powers of 2 it wouldn't be equal zero because we would have 1 in other bites, besides the power of 2, zero we can't consider as a power of 2 because 2**x = 0 - hasn't a solution.

Reply:
PS.  I recommend deleting branches that were merged. 

You can configure autodelete after the merge in the repo.  

![image](https://user-images.githubusercontent.com/171597/111081774-07b4db80-8516-11eb-8ef0-3c467289fdb8.png)